### PR TITLE
Fix Checkstyle Configuration Relative Link

### PR DESCRIPTION
**Pull request by alzimmermsft**
Fixes the relative link to the Checkstyle header file and fixes AppConfiguration Checkstyle issues.

### CODE REVIEW

File: sdk/appconfiguration/azure-data-appconfiguration/README.md [hunk 1]
```diff
@@ -340,11 +340,11 @@ Create a configuration client with proxy options.
 <!-- embedme ./src/samples/java/com/azure/data/appconfiguration/ReadmeSamples.java#L175-L187 -->
 ```Java
 // Proxy options
-final String HOST_NAME = "{your-host-name}";
```
Comment (sima-zhu) [inline sdk/appconfiguration/azure-data-appconfiguration/README.md:343]: Is this fixed by @conniey PR already?

Comment (alzimmermsft) [reply sdk/appconfiguration/azure-data-appconfiguration/README.md:343]: Ah, yes, I see this now, I reviewed that PR before this change was added.

Comment (conniey) [reply sdk/appconfiguration/azure-data-appconfiguration/README.md:343]: Buahaha, guess who gets to fix this merge conflict.

Comment (conniey) [reply sdk/appconfiguration/azure-data-appconfiguration/README.md:343]: What I'm surprised by.. is that the app configuration pipeline didn't catch this.

Comment (alzimmermsft) [reply sdk/appconfiguration/azure-data-appconfiguration/README.md:343]: That is being looked into since it is a severe issue.

**Pull request merged**